File: server/php/src/Application/Exceptions/SdkException.php
<?php

namespace OnlinePayments\ExampleApp\Application\Exceptions;

use Exception;

class SdkException extends Exception
{
    private int $statusCode;

    public function __construct(string $message, int $statusCode, ?Exception $previous = null)
    {
        parent::__construct($message, 0, $previous);
        $this->statusCode = $statusCode;
    }

    public function getStatusCode(): int
    {
        return $this->statusCode;
    }
}

File: server/java/src/main/java/com/onlinepayments/example/service/exceptions/SdkException.java
package com.onlinepayments.example.service.exceptions;

import org.springframework.http.HttpStatus;

public class SdkException extends RuntimeException {
    private final HttpStatus statusCode;

    public SdkException(String message, HttpStatus statusCode) {
        super(message);
        this.statusCode = statusCode;
    }

    public HttpStatus getStatusCode() {
        return statusCode;
    }
}